File: ListNode.cpp
//
//  ListNode.cpp
//  LeetCode
//
//  Created by JackMa on 16/9/14.
//  Copyright © 2016年 JackMa. All rights reserved.
//

#include "ListNode.h"

ListNode *buildList(vector<int> nums) {
    if (nums.size() == 0) {
        return NULL;
    }
    ListNode *root = new ListNode(nums[0]);
    ListNode *p = root;
    
    int i = 1;
    while (i < nums.size()) {
        p->next = new ListNode(nums[i]);
        p = p->next;
        i++;
    }
    
    return root;
}

void outputList(ListNode *head) {
    ListNode *p = head;
    while (p != NULL) {
        cout << p->val << " ";
        p = p->next;
    }
    cout << endl;
}

File: Library/ListNode.swift
//
//  ListNode.swift
//  LeetCode-Swift
//
//  Created by jackma on 2023/3/9.
//  Copyright © 2023 JackMa. All rights reserved.
//

import Foundation

public class ListNode {
    public var val: Int
    public var next: ListNode?
    
    public convenience init() {
        self.init(0, nil)
    }
    
    public convenience init(_ val: Int) {
        self.init(val, nil)
    }
    
    public init(_ val: Int, _ next: ListNode?) {
        self.val = val
        self.next = next
    }
}

extension ListNode {
    
    /// Build a list with nums
    /// - Parameter nums: nums
    /// - Returns: list head
    static func buildListNode(_ nums: [Int]) -> ListNode? {
        var root = ListNode()
        
        var curr = root
        for num in nums {
            let node = ListNode(num)
            curr.next = node
            if let next = curr.next {
                curr = next
            }
        }
        
        return root.next
    }
    
}
